version-ranges/src/lib.rs

```diff
@@ -283,6 +290,37 @@ impl<V: Ord> Ranges<V> {
         }
     }
 
+    /// We want to use `iterator_try_collect`, but since it's unstable at the time of writing,
+    /// we expose a public `FromIterator<(Bound<V>, Bound<V>)>` method and use this for internal
+    /// testing.
+    fn try_from(
```

Comment:
when I see `try_from` I think of the `TryFrom` trait. Perhaps `try_from_iterator`?